module/os/linux/zfs/zpl_super.c

```diff
@@ -117,8 +154,25 @@ zpl_sync_fs(struct super_block *sb, int wait)
 	error = -zfs_sync(sb, wait, cr);
 	spl_fstrans_unmark(cookie);
 	crfree(cr);
-	ASSERT3S(error, <=, 0);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
+#ifdef HAVE_SUPER_BLOCK_S_WB_ERR
+	if (error && wait)
+		errseq_set(&sb->s_wb_err, error);
+#else
+	if (error && wait) {
+		zfsvfs_t *zfsvfs = sb->s_fs_info;
+		ASSERT3P(zfsvfs, !=, NULL);
+		if (zfs_enter(zfsvfs, FTAG) == 0) {
+			txg_wait_synced(dmu_objset_pool(zfsvfs->z_os), 0);
```

Comment:
I worry that blocking system-wide `sync` calls might be too disruptive.  I'd say: "if they don't care about errors, why should we?".

Reply:
Yeah...